Game over

poltak opened the pull request:
Fixes #31

poltak committed:
Normalize `fallingObjects` state to be object indexed by ID

- previously was an array, which is indexed by insertion order, so we have to loop over it whenever we need to perform a mutation on a specific object
- "normalizing" it allows us to index it by an ID meaningful to us, so we can always access specific objects directly using their IDs
- also removed update falling obj numerator/denominator mutations, as they aren't used and unlikely they will be

poltak committed:
Add gitignore file - remove .DS_Store

Commit:
Implement level, fix #32

poltak committed:
Merge branch 'feature/Implement-level-and-fix-32' into feature/game-over

# Conflicts:
#	falling.js
#	index.html
#	stateMutators.js

poltak committed:
Update calcLives mutator to set game over if < 1 life

poltak committed:
Set up event handler to restart game

- TODO: restart falling algo

tranhuuphucuong commented:
Now this is perfect..... I like it @poltak 
f4da804f2ef0f5db342f8d5ed69a15915ec3f5ae

poltak committed:
Fix overlay render function

- it wasn't properly taking into account state changes. So it would set display to 'block' on one text, but when the state changed to show another text it wouldn't set the prev one back to 'none'. So you would end up with 2 text el's visible
- now it properly updates all text elements each render call

poltak committed:
Allow game to restart from game over mode

poltak committed:
Rename `update` mutator to `catchFallingObject`

- more descriptive

tranhuuphucuong reviewed (approved):
Everything looks good I will merge it and fix the bug about removing object later
